### ISSUE TITLE
Bootstrap show wrong media query on Android

### ISSUE THREAD

**Issue by mewsoft** [twbs/bootstrap#15549]
I have this test code:
```
    <!doctype html>
    <html lang="en">
     <head>
      <meta charset="UTF-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
        <meta name="viewport" content='width=device-width, initial-scale=1.0, user-scalable=yes' />

      <link href="//maxcdn.bootstrapcdn.com/bootstrap/3.3.1/css/bootstrap.min.css" rel="stylesheet">
      <script type="text/javascript" src="//ajax.googleapis.com/ajax/libs/jquery/1.10.1/jquery.min.js"></script>
      <script src="//maxcdn.bootstrapcdn.com/bootstrap/3.3.1/js/bootstrap.min.js"></script>

      <title>Bootstrap Media Query Check</title>

     </head>
     <body>
      
    
    <div class="container-fluid">    
        <div class="row">
            <div class=" visible-xs alert-success">XS </div>
            <div class=" visible-sm alert-danger">SM </div>
            <div class=" visible-md alert-warning">MD </div>
            <div class=" visible-lg alert-info">LG </div>
    	</div>      
    </div>
    
     </body>
    </html>
```

When I run this on Android 4.x device and its resolution is 540x960, the code
shows `XS` in both mobile orientations.

Is it supposed to show `XS` and `SM` in vertical and horizontal directions.

On PC it shows the correct sizes.

What I am missing?.

**Comment by cvrebert** [comment]
Duplicate of http://stackoverflow.com/questions/27906691/bootstrap-show-wrong-media-query-on-android

**Comment by cvrebert** [comment]
JS Bin: http://jsbin.com/zifiqi/1/edit

**Comment by twbs-lmvtfy** [comment]
Hi @cvrebert!

You appear to have posted a live example (http://jsbin.com/zifiqi/1/edit), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* line 20, column 13: [E013](https://github.com/twbs/bootlint/wiki/E013): Only columns (`.col-*-*`) may be children of `.row`s
* line 21, column 13: [E013](https://github.com/twbs/bootlint/wiki/E013): Only columns (`.col-*-*`) may be children of `.row`s
* line 22, column 13: [E013](https://github.com/twbs/bootlint/wiki/E013): Only columns (`.col-*-*`) may be children of `.row`s
* line 23, column 13: [E013](https://github.com/twbs/bootlint/wiki/E013): Only columns (`.col-*-*`) may be children of `.row`s
* [W002](https://github.com/twbs/bootlint/wiki/W002): `<head>` is missing X-UA-Compatible `<meta>` tag that disables old IE compatibility modes

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by cvrebert** [comment]
After pedantic adjustments: http://jsbin.com/zifiqi/3/edit

**Comment by cvrebert** [comment]
Doesn't repro in Chrome on Android 5.
Are you using Chrome or the Android "Browser" app?

**Comment by mewsoft** [comment]
Your  http://jsbin.com/zifiqi/3/edit just outputs xs on both direction on the mobile Android 4.2.2 resolution 540x960.
My second question to make sure I understand correct or not, when I flip the phone to the 960pixel should the layout switch to sm and I understood? or the bootstrap always depends on the physical device vertical width?

**Comment by cvrebert** [comment]
It uses the viewport width, which should change when you rotate your device, as opposed to the deprecated physical device width.
What does http://ryanve.com/lab/dimensions/ report on your devices?

> outputs xs on both direction on the mobile Android 4.2.2 resolution 540x960

Your original post said that it output SM. Which is it? XS or SM?

**Comment by mewsoft** [comment]
Sorry for the typo, it shows `XS` in both directions.
I just tried this URL  http://ryanve.com/lab/dimensions/ 
and it shows in vertical direction 360 width x 615 height
and in horizontal direction  640 width x 355 height
This device is Huawi G630, I will try Another device Sony XP with 720x1280 and let you know

**Comment by mewsoft** [comment]
This link  http://ryanve.com/lab/dimensions/  shows for Sony Xperia XP 720x120 these dimensions
Vertical: 360x567
Horizontal:  598x279

**Comment by cvrebert** [comment]
So, your viewports are indeed XS.
SM would be a minimum viewport width of 768px, but your reported numbers are all smaller than that. Remember that physical screen width != viewport width due to the browser's address bar etc. and Android's system status bars, toolbars etc.

**Comment by mewsoft** [comment]
Are you saying a Mobile with 1280x720 pixel will be usable as  598x279 only. Nothing is used on left and right to eat all this. I search online and all reports said the jquery .width functions are not correct.
Why you are closing the topic without a real solution.

**Comment by mewsoft** [comment]
This URL http://ipad.atwebpages.com/viewport.html shows:
Your visual viewport(widthxheight) is: 980 x 420
Your layout viewport (widthxheight) is: 980 x 427
Your device-width x height is: 360 x 640;

these numbers for Huawi device 540x960

**Comment by cvrebert** [comment]
You need to use a test page with a viewport meta tag, e.g. http://ipad.atwebpages.com/viewport2.html

**Comment by mewsoft** [comment]
With that meta tag numbers are different:

Your visual viewport(widthxheight) is: 598 x 279
Your layout viewport (widthxheight) is: 598x279
Your device-width x height is: 640x360;

Does that means If I remove the meta tag from the page will fix it?

**Comment by cvrebert** [comment]
You can learn more about the viewport meta tag here:
https://developers.google.com/web/fundamentals/layouts/rwd-fundamentals/set-the-viewport?hl=en
